server/src/main/java/com/cloud/network/element/VirtualRouterElement.java

```diff
@@ -767,7 +767,8 @@ public boolean saveSSHKey(final Network network, final NicProfile nic, final Vir
 
     @Override
     public boolean saveHypervisorHostname(NicProfile nicProfile, Network network, VirtualMachineProfile vm, DeployDestination dest) throws ResourceUnavailableException {
-        if (_networkModel.getUserDataUpdateProvider(network).getProvider().equals(Provider.VirtualRouter) && vm.getVirtualMachine().getType() == VirtualMachine.Type.User) {
+        final UserDataServiceProvider userDataUpdateProvider = _networkModel.getUserDataUpdateProvider(network);
+        if (userDataUpdateProvider != null && Provider.VirtualRouter.equals(userDataUpdateProvider.getProvider()) && vm.getVirtualMachine().getType() == VirtualMachine.Type.User) {
```

Comment:
Should we return false if userdata provider is null?  This will atleast log the appropriate error message.

Reply:
@sureshanaparti can you answer this query?

Reply:
addressed, thanks @Pearl1594